packages/zoe/src/zoe.js

```diff
@@ -66,47 +69,54 @@ import { makeTables } from './state';
- * with keywords as keys. Terms are the arguments to the contract,
+ * Terms are the arguments to the contract,
  * such as the number of bids an auction will wait for before closing.
  * Terms are up to the discretion of the smart contract. We get back
  * an invite (an ERTP payment) to participate in the contract.
  *
- * TODO: property {(instanceHandle: InstanceHandle) => InstanceRecord} getInstanceRecord
+ * @property {(InstanceHandle) => InstanceRecord} getInstanceRecord
  * Credibly get information about the instance (such as the installation
  * and terms used).
  *
- * @property {(invite: Invite, proposal: Proposal, payments: PaymentKeywordRecord) => Promise<SeatAndPayout>} redeem
+ * @property {(invite: Invite,
+ *             proposal: Proposal?,
+ *             paymentKeywordRecord: PaymentKeywordRecord?)
+ *            => Promise<OfferResultRecord>} offer
  * To redeem an invite, the user normally provides a proposal (their rules for the
  * offer) as well as payments to be escrowed by Zoe.  If either the proposal or payments
  * would be empty, indicate this by omitting that argument or passing undefined, rather
  * than passing an empty record.
  *
  * The proposal has three parts: `want` and `give` are used
  * by Zoe to enforce offer safety, and `exit` is used to specify
- * the extent of payout liveness that Zoe can guarantee.
+ * the particular payout-liveness policy that Zoe can guarantee.
  * `want` and `give` are objects with keywords as keys and amounts
- * as values. `payments` is a record with keywords as keys,
+ * as values. `paymentKeywordRecord` is a record with keywords as keys,
  * and the values are the actual payments to be escrowed. A payment
  * is expected for every rule under `give`.
  *
+ * @property {(offerHandle: OfferHandle) => boolean} isOfferActive
+ * @property {(offerHandles: OfferHandle[]) => Offer[]} getOffers
+ * @property {(offerHandle: OfferHandle) => Offer} getOffer
  * @property {(installationHandle: InstallationHandle) => InstallationRecord} getInstallation
  *
- * @typedef {Object} SeatAndPayout This is returned by a call to `redeem` on Zoe.
+ * @typedef {Object} ContractAPI
+ * A contract-specific value that is returned by the OfferHook.
+ * TODO: 'ContractAPI' is a bad name, maybe more like 'OfferReturnValue'
  *
- * @property {Object} seat An arbitrary object whose methods allow the user to take
- * certain actions in a contract. The payout is a promise that resolves
- * to an object which has keywords as keys and promises for payments
- * as values. Note that while the payout promise resolves when an offer
- * is completed, the promise for each payment resolves after the remote
- * issuer successfully withdraws the payment.
  *
- * @property {Payment[]} payout A promise that resolves
- * to an object which has keywords as keys and promises for payments
+ * @typedef {Object} OfferResultRecord This is returned by a call to `offer` on Zoe.
```

Comment:
The definition of `offer` about 17 lines earlier mentions that:

```js
 * @property {(invite: Invite,
 *             proposal: Proposal?,
 *             paymentKeywordRecord: PaymentKeywordRecord?)
 *            => Promise<OfferResultRecord>} offer
 * To redeem an invite, the user normally provides a proposal (their rules for the
 * offer) as well as payments to be escrowed by Zoe.  If either the proposal or payments
 * would be empty, indicate this by omitting that argument or passing undefined, rather
 * than passing an empty record.
```

Is that sufficient? I think we're defining the signature of `offer` here, in the signature of `ZoeService`, rather than down at the definition of `offer` itself, which means we're using the fat-arrow style of JSdoc rather than the `function()` style (if that distinction makes sense).

Reply:
I don't know. @michaelfig , opinion?

Reply:
The `@callback` syntax allows you to annotate the individual arguments.  Otherwise, no, I don't have an opinion.

Reply:
Oh, and @warner, it should be:
```
      proposal?: Proposal,
```

The optional argument has a question mark on the parameter, not on the type.

Reply:
I've just fixed the question marks (in both `proposal` and `paymentKeywordRecord`) in cd91d06d.

Should I change the other part? Or leave it alone?

Reply:
@michaelfig do we prefer square brackets or question marks? I thought you advised square brackets. We should decide on a convention and stick to it.

Nevertheless, resolving this conversation because it is now correct. Deciding on a company convention can happen later.

Reply:
Assuming "leave it alone" is fine, so resolving.